controllers/managedocs_controller.go

```diff
@@ -518,76 +444,107 @@ func (r *ManagedOCSReconciler) reconcileAlertmanager() error {
 	return nil
 }
 
-func (r *ManagedOCSReconciler) reconcileAlertmanagerConfig() error {
+type smtpConfig struct {
+	Endpoint           string   `yaml:"endpoint"`
+	Username           string   `yaml:"username"`
+	Password           string   `yaml:"password"`
+	FromAddress        string   `yaml:"fromAddress"`
+	NotificationEmails []string `yaml:"notificationEmails"`
+}
+
+type pagerDutyConfig struct {
+	SOPEndpoint string `yaml:"sopEndpoint"`
+	SecretKey   string `yaml:"secretKey"`
+}
```

Comment:
- I remember seeing this same signature in one of previous commits
- If this is one-to-one mapping to CRD spec, any reason not using the same by just adding `yaml` tag to struct fields?
- Or if these are removed from the spec, pls disregard this comment